src/many-sources/abstract-pool.js

```diff
@@ -34,8 +35,15 @@ inherit(AbstractPool, Stream, {
       if (this._queueLim === -1 || this._queue.length < this._queueLim) {
         this._addToQueue(toObs(obj));
       } else if (this._drop === 'old') {
-        this._removeOldest();
-        this._add(obj, toObs);
+        if (this._overlapping) {
+          this._addToCur(toObs(obj));
+          while (this._curSources.length > this._concurLim) {
+            this._removeOldest();
+          }
```

Comment:
this loop seemed a little weird when I was working on it, however, I think to have correct parity with the other side of the condition it needs to be here. Given that `this._curSources.length - this._concurLim > 1`, the other side, implicitly loops due to recursion. Since we want to add _first_, using `_addToCur`, then we need to make sure that we aren't over the `_concurLim` and blindly removing one might be not enough. Although, I'm not sure how an `AbstractPool` could get into that state.